test/common-tests/request-query-validation-api-tests/string-query-validation-api-tests.ts

```diff
@@ -0,0 +1,194 @@
+import { getMinAndMaxLengthFromOptions } from '@ukef-test/support/helpers/min-and-max-length-helper';
+import request from 'supertest';
+
+export interface StringQueryValidationApiTestOptions<RequestQueryItems, RequestQueryItemsKey extends keyof RequestQueryItems> {
+  queryName: RequestQueryItemsKey;
+  length?: number;
+  minLength?: number;
+  maxLength?: number;
+  required?: boolean;
+  pattern?: RegExp;
+  enum?: any;
+  generateQueryValueOfLength?: (length: number) => RequestQueryItems[RequestQueryItemsKey];
+  generateQueryValueThatDoesNotMatchRegex?: () => RequestQueryItems[RequestQueryItemsKey];
+  generateQueryValueThatDoesNotMatchEnum?: () => RequestQueryItems[RequestQueryItemsKey];
+  validRequestQueries: RequestQueryItems;
+  makeRequestWithQueries: (queries: RequestQueryItems) => request.Test;
+  givenAnyRequestQueryWouldSucceed: () => void;
+}
+
+export const withStringQueryValidationApiTests = <RequestQueryItems, RequestQueryItemsKey extends keyof RequestQueryItems>({
+  queryName: queryNameSymbol,
+  length: lengthOption,
+  minLength: minLengthOption,
+  maxLength: maxLengthOption,
+  required,
+  pattern,
+  enum: theEnum,
+  generateQueryValueOfLength,
+  generateQueryValueThatDoesNotMatchRegex,
+  generateQueryValueThatDoesNotMatchEnum,
+  validRequestQueries,
+  makeRequestWithQueries,
+  givenAnyRequestQueryWouldSucceed,
+}: StringQueryValidationApiTestOptions<RequestQueryItems, RequestQueryItemsKey>) => {
+  const queryName = queryNameSymbol as string;
+  const { minLength, maxLength } = getMinAndMaxLengthFromOptions({ parameterName: queryName, minLengthOption, maxLengthOption, lengthOption });
+
+  required = required ?? true;
+
+  describe(`${queryName} validation`, () => {
+    beforeEach(() => {
+      givenAnyRequestQueryWouldSucceed();
+    });
+
+    it('returns a 2xx response if the query parameter is valid', async () => {
+      const { status } = await makeRequestWithQueries(validRequestQueries);
+
+      expect(status).toBeGreaterThanOrEqual(200);
+      expect(status).toBeLessThan(300);
+    });
+
+    it('returns a 2xx response if there is an unexpected query parameter', async () => {
+      const { status } = await makeRequestWithQueries({ ...validRequestQueries, unexpectedQueryParameter: 'unexpectedQueryParameterValue' });
+
+      expect(status).toBeGreaterThanOrEqual(200);
+      expect(status).toBeLessThan(300);
+    });
+
+    it(`returns a 400 response if ${queryName} is number`, async () => {});
```

Comment:
We've missed out a test here

Reply:
Yep -- We don't need one here I believe as it transforms the number to a string